plugins/woocommerce/includes/class-wc-shipping-rate.php

```diff
@@ -232,6 +246,15 @@ public function get_shipping_tax() {
 		return apply_filters( 'woocommerce_get_shipping_tax', count( $this->taxes ) > 0 && ! WC()->customer->get_is_vat_exempt() ? (float) array_sum( $this->taxes ) : 0.0, $this );
 	}
 
+	/**
+	 * Get tax status.
+	 *
+	 * @return string
+	 */
+	public function get_tax_status() {
+		return apply_filters( 'woocommerce_shipping_rate_tax_status', $this->data['tax_status'], $this );
```

Comment:
It doesn't seem like we're going to support other tax statuses "soon" (other than `none` or `taxable`). As such, I wonder if we should sanitize the value here (after the filter is applied) to prevent incorrect values from bubbling up.